src/cata_tiles.h

```diff
@@ -25,9 +25,37 @@ class JsonObject;
 struct visibility_variables;
 
 /** Structures */
+struct sprite_variation {
+    int weight;
+    std::vector<int> sprite_ids;
+
+    sprite_variation()
+    {
+        weight = 1;
```

Comment:
"empty" and "invalid" are two different things. empty lists can get populated during the loading process. and items can be removed.

the -1 "invalid" marker is for exactly one case, the first variation in each list of variations. I *could* just start all of the loops at 1, skipping the 0th element, and then not bother with a weight on that variation, but that seems weirder.